### PR TITLE
feat(hog): fix elements matching

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

It started from this:
image

This exposes two problems: 
- we don't have the new materialized `elements_chain_*` columns in Hog globals
- we can't run functions that use lambdas, like `arrayExists` (solved)

## Changes

- Update the "expected volume" calculation to take into account global properties (those not on any one series)
- Remove some unused code from the plugin server (`ParsedClichouseEvent` was not used anywhere 🤷)
- Add the 4 new fields `elements_chain_href`, `elements_chain_texts`, `elements_chain_ids`, `elements_chain_elements` into Hog filter matching. 
- These fields are stored in the database as materialized columns. We use the same regex that's used by ClickHouse to calculate them in the plugin server.
- The fields are added lazily, meaning they're calculated only when accessed, or when the bytecode is serialized.
- Adds functions `indexOf`, `position`, `positionCaseInsensitive` and `arrayCount`

## Extra

- I'd like to add "elements_chain" higher up on the `HogFunctionInvocationGlobals` field `event`, in order to better match the actual database schema. However since it has more discrepancies (`name` instead of `event` for the actual event, etc), I decided to leave it for now.

## Alternatives considered

- Always calculating these materialized columns into `event` instead of lazily getting them. Rejected as this will be wasteful if you have no functions that look for element texts
- Moving the regex calculation into Hog. This is a lot more work as Hog doesn't (yet) support lazy properties, and I'd then have to inject some "global hog" code before each filter.

## How did you test this code?

Added a test in `hog-executor.test.ts`, poked around in the UI.

### CODE REVIEW

File: plugin-server/src/cdp/utils.ts [hunk 1]
```diff
@@ -114,14 +134,36 @@ export function convertToHogFunctionFilterGlobal(globals: HogFunctionInvocationG
...
+        elements_chain_elements: [],
         timestamp: globals.event.timestamp,
         properties: globals.event.properties,
         person: globals.person ? { properties: globals.person.properties } : undefined,
         ...groups,
     }
+    if (elementsChain) {
+        Object.defineProperties(response, {
+            elements_chain_href: {
+                get: () => getElementsChainHref(elementsChain),
```
Comment (benjackwhite) [inline plugin-server/src/cdp/utils.ts:153]: Love this solution. I'm wondering though if we should also cache these values? Maybe not worth it for now...

Also a comment here would be helpful for future travellers to understand what this is all for (I barely get it right now)

Comment (benjackwhite) [reply plugin-server/src/cdp/utils.ts:153]: Also its not awesome that there is only a test for one path...

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: plugin-server/src/cdp/utils.ts (hunk 1 above)
Comment (mariusandra) [reply plugin-server/src/cdp/utils.ts]: I think I found a rather neat way to cache this: https://github.com/PostHog/posthog/pull/24331/commits/e2e1bc73721ad764b9cf9bd63982c1123200886f

More tests coming soon...

**Comment by mariusandra** [comment]
Ready for a re-review

File: plugin-server/src/cdp/utils.ts (hunk 1 above)
Comment (benjackwhite) [reply plugin-server/src/cdp/utils.ts]: wooohhhhhh self destructing function :o

**Review by benjackwhite — APPROVED** [review]

**Pull request merged**